react/AppLinker/index.jsx

```diff
@@ -0,0 +1,131 @@
+/* global __TARGET__ */
+
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import {
+  checkApp,
+  startApp,
+  isMobileApp,
+  isMobile,
+  openDeeplinkOrRedirect
+} from 'cozy-device-helper'
+import { NATIVE_APP_INFOS } from 'cozy-ui/transpiled/react/AppLinker/native'
+
+import expiringMemoize from './expiringMemoize'
+
+const expirationDelay = 10 * 1000
+const memoizedCheckApp = expiringMemoize(
+  checkApp,
+  expirationDelay,
+  appInfo => appInfo.id
+)
+
+/**
+ * Render-props component that provides onClick/href handler to
+ * apply to an anchor that needs to open an app.
+ *
+ * Handles several cases:
+ *
+ * - On mobile app and other mobile app available
+ * - On web (not mobile)
+ * - On web mobile
+ *
+ * @example
+ * ```
+ * <AppLinker slug='banks' href='http://dalailama-banks.mycozy.cloud'>{
+ *   ({ onClick, href, name }) => {
+ *     <a href={href} onClick={onClick}>
+ *       Open { name }
+ *     </a>
+ *   }
+ * }</AppLinker>
+ * ```
+ */
+export class AppLinker extends React.Component {
+  state = {
+    isMobileAppAvailable: null
+  }
+
+  constructor() {
+    super()
+    this.openNativeApp = this.openNativeApp.bind(this)
+  }
+
+  componentDidMount() {
+    if (__TARGET__ === 'mobile') {
+      this.checkAppAvailability()
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.switchTimeout) clearTimeout(this.switchTimeout)
+  }
+
+  async checkAppAvailability() {
+    const { slug } = this.props.app
+    const appInfo = NATIVE_APP_INFOS[slug]
+    if (appInfo) {
+      const isMobileAppAvailable = Boolean(await memoizedCheckApp(appInfo))
+      this.setState({ isMobileAppAvailable })
+    }
+  }
+
+  onAppSwitch = () => {
+    const { onAppSwitch } = this.props
+    if (typeof onAppSwitch === 'function') {
+      this.switchTimeout = setTimeout(() => {
```

Comment:
The delay before the app switch comes from the cozy-bar, but I'm not sure we should have it here? This seems to be a UX thing specific to cozy-bar, no?

Reply:
You're right, I am removing the setTimeout logic and delay, it will be the responsibility of the app.